REPLICATION_POLICIES.md

```diff
@@ -0,0 +1,293 @@
+<!-- We don't want the device to go to sleep while a replication is in progress. -->
+<uses-permission android:name="android.permission.WAKE_LOCK" />
+<!-- We want to restart our periodic replications after a reboot. -->
+<uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
+```
+
+###Binding to the Service
+Lets assume we have an `Activity` that displays our data to the user. While this `Activity` is displayed we want our more frequent updates (every 5 minutes), and we want to know when replication has completed so we can refresh the Activity's UI with the new data. Note that the `Activity` must be running in the same process as the `Service`, which is the default on Android.
+
+First we add some fields to our `Activity`:
+```java
+// Add a handler to allow us to post UI updates on the main thread.
+private final Handler mHandler = new Handler(Looper.getMainLooper());
+
+// Reference to our service.
+private ReplicationService mReplicationService;
+
+// Flag indicating whether the Activity is currently bound to the Service.
+private boolean mIsBound;
+```
+
+Now we add a `ServiceConnection` to our `Activity` to allow us to handle binding and unbinding from our `MyReplicationService`. This enables us to get a reference to the service when we bind to it and add a listener for `replicationComplete` to the `Service`:
+
+```java
+private ServiceConnection mConnection = new ServiceConnection() {
+    @Override
+    public void onServiceConnected(ComponentName name, IBinder service) {
+        mReplicationService = ((ReplicationService.LocalBinder) service).getService();
+        mReplicationService.addListener(new ReplicationService.SimpleReplicationCompleteListener() {
+            @Override
+            public void replicationComplete(int id) {
+                // Check if this is the pull replication
+                if (id == MyReplicationService.PULL_REPLICATION_ID) {
+                    mHandler.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            // Update the UI.
+                        }
+                    });
+                }
+            }
+
+            @Override
+            public void replicationErrored(int id) {
+                // We'll just ignore the error.
+            }
+        });
+    }
+
+    @Override
+    public void onServiceDisconnected(ComponentName name) {
+        mReplicationService = null;
+    }
+};
+```
+We now bind to the service in `Activity.onStart()` and unbind in `Activity.onStop()`:
+
+```java
+@Override
+protected void onStart() {
+    super.onStart();
+    bindService(new Intent(this, MyReplicationService.class), mConnection, Context.BIND_AUTO_CREATE);
+    mIsBound = true;
+}
+
+@Override
+protected void onStop() {
+    super.onStop();
+    if (mIsBound) {
+        unbindService(mConnection);
+        mIsBound = false;
+    }
+}
+```
+
+# Java replication policies
```

Comment:
I wonder if we could shift this up to the top, as it's so short. The counter-argument is that this is the less-used case.